include/pybind11/embed.h

```diff
@@ -146,9 +115,64 @@ inline void initialize_interpreter(bool init_signal_handlers = true,
         pybind11_fail("The interpreter is already running");
     }
 
+#if PY_VERSION_HEX < 0x030B0000
+
     Py_InitializeEx(init_signal_handlers ? 1 : 0);
 
-    detail::set_interpreter_argv(argc, argv, add_program_dir_to_path);
+    // Before it was special-cased in python 3.8, passing an empty or null argv
+    // caused a segfault, so we have to reimplement the special case ourselves.
+    bool special_case = (argv == nullptr || argc <= 0);
+
+    const char *const empty_argv[]{"\0"};
+    const char *const *safe_argv = special_case ? empty_argv : argv;
+    if (special_case) {
+        argc = 1;
+    }
+
+    auto argv_size = static_cast<size_t>(argc);
+    // SetArgv* on python 3 takes wchar_t, so we have to convert.
+    std::unique_ptr<wchar_t *[]> widened_argv(new wchar_t *[argv_size]);
+    std::vector<std::unique_ptr<wchar_t[], detail::wide_char_arg_deleter>> widened_argv_entries;
+    widened_argv_entries.reserve(argv_size);
+    for (size_t ii = 0; ii < argv_size; ++ii) {
+        widened_argv_entries.emplace_back(detail::widen_chars(safe_argv[ii]));
+        if (!widened_argv_entries.back()) {
+            // A null here indicates a character-encoding failure or the python
+            // interpreter out of memory. Give up.
+            return;
+        }
+        widened_argv[ii] = widened_argv_entries.back().get();
+    }
+
+    auto *pysys_argv = widened_argv.get();
+
+    PySys_SetArgvEx(argc, pysys_argv, static_cast<int>(add_program_dir_to_path));
+#else
+    PyConfig config;
```

Comment:
A comment here would be useful (similar to Henry's comment from a few days ago):

* Link to the corresponding Python sources.
* Something like "unfortunately the Python code is private, therefore we are forced to duplicate here".

Reply:
What has to be duplicated?

Reply:
PyConfig API initializes the default value of sys.path. But Py_RunMain() then inserts a path to sys.path. Python 3.11 safe_path=1 disables this behavior:
* https://docs.python.org/dev/c-api/init_config.html#c.PyConfig.safe_path
* https://docs.python.org/dev/c-api/init_config.html#python-path-configuration

It might be interesting to move more code changing sys.path into the Python initialization (PyConfig), but it's complicated.